src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,196 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private void validateIndexExclusive(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Wrong index provided");
+        }
+    }
+
+    private void validateIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Wrong index provided");
+        }
+    }
+
+    private Node getNode(int index) {
+        validateIndex(index);
+        if (index == 0) {
+            return head;
+        } else if (index == size - 1) {
+            return tail;
+        }
+        Node currentNode;
+        int steps = stepsToTake(index);
+        currentNode = steps > 0 ? head : tail;
+
+        // Should I reverse if and for order ?
+        // It will add to readability, but it will execute a bit longer
+        if (steps > 0) {
+            for (int i = 0; i < steps; i++) {
+                currentNode = currentNode.next;
+            }
+        } else {
+            // Should change steps sign to plus
+            for (int i = 0; i < -steps; i++) {
+                currentNode = currentNode.prev;
+            }
+        }
```

Comment:
Confusing second loop. 
Consider backward iteration, e.g. 
```
for (int i = lastIndex; i != -1; i--)
```
My point, is better to use while loop here, because we have linked nodes

```suggestion
        if (steps > 0) {
            for (int i = 0; i < steps; i++) {
                currentNode = currentNode.next;
            }
        } else {
            // Should change steps sign to plus
            for (int i = 0; i < -steps; i++) {
                currentNode = currentNode.prev;
            }
        }
```